stdlib/json.gr

```diff
@@ -2069,3 +2069,298 @@ provide let parse: (str: String) => Result<Json, JsonParseError> = (str: String)
+   */
+  provide let map = (lens, fn, subject) => {
+    match (lens.get(subject)) {
+      Some(lensVal) => lens.set(fn(lensVal), subject),
+      None => None,
+    }
+  }
+
+  /**
+   * A lens whose focus is a JSON value.
+   * 
+   * @example assert get(json, JsonString("abc")) == Some(JsonString("abc"))
+   * 
+   * @since v0.7.0
+   */
+  provide let json = {
+    get: json => Some(json),
+    set: (newValue, _) => Some(newValue),
+  }
+
+  /**
+   * A lens whose focus is a JSON boolean value.
+   * 
+   * @example assert get(boolean, JsonBoolean(true)) == Some(true)
+   * 
+   * @since v0.7.0
+   */
+  provide let boolean = {
+    get: json => {
+      match (json) {
+        JsonBoolean(val) => Some(val),
+        _ => None,
+      }
+    },
+    set: (newValue, _) => Some(JsonBoolean(newValue)),
+  }
+
+  /**
+   * A lens whose focus is a JSON string value.
+   * 
+   * @example assert get(string, JsonString("abc")) == Some("abc")
+   * 
+   * @since v0.7.0
+   */
+  provide let string = {
+    get: json => {
+      match (json) {
+        JsonString(val) => Some(val),
+        _ => None,
+      }
+    },
+    set: (newValue, _) => Some(JsonString(newValue)),
+  }
+
+  /**
+   * A lens whose focus is a JSON number value.
+   * 
+   * @example assert get(number, JsonNumber(123)) == Some(123)
+   * 
+   * @since v0.7.0
+   */
+  provide let number = {
+    get: json => {
+      match (json) {
+        JsonNumber(val) => Some(val),
+        _ => None,
+      }
+    },
+    set: (newValue, _) => Some(JsonNumber(newValue)),
+  }
+
+  /**
+   * A lens whose focus is a JSON array.
+   * 
+   * @example assert get(array, JsonArray([JsonNumber(123)])) == Some([JsonNumber(123)])
+   * 
+   * @since v0.7.0
+   */
+  provide let array = {
+    get: json => {
+      match (json) {
+        JsonArray(val) => Some(val),
+        _ => None,
+      }
+    },
+    set: (newValue, _) => Some(JsonArray(newValue)),
+  }
+
+  /**
+   * A lens whose focus is the properties of a JSON object.
+   * 
+   * @example assert get(objectProps, JsonObject([("a", JsonNumber(123))])) == Some([("a", JsonNumber(123))])
+   * 
+   * @since v0.7.0
+   */
+  provide let objectProps = {
+    get: json => {
+      match (json) {
+        JsonObject(val) => Some(val),
+        _ => None,
+      }
+    },
+    set: (newValue, _) => Some(JsonObject(newValue)),
+  }
+
+  /**
+   * Creates a lens whose focus is a given property of a JSON object.
+   * 
+   * @param propertyName: The property name of the JSON object to focus on
+   * @returns A lens whose focus is the given property of a JSON object
+   * 
+   * @example assert get(prop("x"), JsonObject([("x", JsonNumber(123))])) == Some(JsonNumber(123))
+   * @example
+   * assert set(prop("x"), JsonString("new"), JsonObject([("x", JsonNumber(123))])) ==
+   *   Some(JsonObject([("x", JsonString("new"))]))
+   * 
+   * @since v0.7.0
+   */
+  provide let prop = propertyName =>
+    { get: json => match (json) {
```

Comment:
Unrelated to this PR but I don't really like how the formatter handles these records here